src/Autocomplete/doc/index.rst

```diff
@@ -171,6 +171,109 @@ After creating this class, use it in your form:
+while adding a `food` field to the `FoodForm`. However, when the Ajax is used to fetch the results, on the consequent renders,
+the default `query_builder` will be used::
+
+    // src/Form/FoodForm.php
+    // ...
+
+    class FoodForm extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options): void
+        {
+            $currentFoodId = $builder->getData()->getId();
+
+            $builder
+                ->add('food', FoodAutocompleteField::class, [
+                    'query_builder' => function (EntityRepository $er) {
+                            $qb = $er->createQueryBuilder('o');
+
+                            $qb->andWhere($qb->expr()->notIn('o.id', [$currentFoodId]));
+
+                            return $qb;
+                        };
+                    }
+                ])
+            ;
+        }
+    }
+
+If some food can be consisted of other foods, we might want to exclude the "root" food from the list of available foods.
+To achieve this, we can remove the `query_builder` option from the above example and pass the `excluded_foods` extra option
+to the `FoodAutocompleteField`::
+
+    // src/Form/FoodForm.php
+    // ...
+
+    class FoodForm extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options): void
+        {
+            $currentFoodId = $builder->getData()->getId();
+
+            $builder
+                ->add('food', FoodAutocompleteField::class, [
+                    'extra_options' => [
+                        'excluded_foods' => [$currentFoodId],
+                    ],
+                )
+            ;
+        }
+    }
+
+The magic of the `extra_options` option is that it will be passed to the `FoodAutocompleteField` every time an Ajax call is made.
+So now, we can just use the `excluded_foods` extra option in the default `query_builder` of the `FoodAutocompleteField`::
+
+    // src/Form/FoodAutocompleteField.php
+    // ...
+
+    use Symfony\Bundle\SecurityBundle\Security;
+    use Symfony\UX\Autocomplete\Form\AsEntityAutocompleteField;
+    use Symfony\UX\Autocomplete\Form\BaseEntityAutocompleteType;
+
+    #[AsEntityAutocompleteField]
+    class FoodAutocompleteField extends AbstractType
+    {
+        public function configureOptions(OptionsResolver $resolver)
+        {
+            $resolver->setDefaults([
+                // ...
+                'query_builder' => function (Options $options) {
+                    return function (EntityRepository $er) use ($options) {
+                        $qb = $er->createQueryBuilder('o');
+
+                        $excludedFoods = $options['extra_options']['excluded_foods'] ?? [];
+                        if ([] !== $excludedFoods) {
+                            $qb->andWhere($qb->expr()->notIn('o.id', $excludedFoods));
+                        }
+
+                        return $qb;
+                    };
+                }
```

Comment:
I love this example because it's very relatable. 

On the other side, i'm not sure we want to push that much the "query_builder" option, as it may be hard to grasp for newcomers 

Just wondering if we could illustrate the extra_options with something "simpler" ... 
.. if not this example is great :)